virtualenv_tools.py

```diff
@@ -91,17 +91,25 @@ def update_script(
             return
         f.seek(0)
         lines = list(f)
-    args = lines[0][2:].strip().split()
+
+    if lines[0].strip() == b'#!/bin/sh' and lines[1].startswith(b"'''exec' "):
+        loc = 1
+    else:
+        loc = 0
+
+    args = lines[loc][2:].strip().split()
+    first_two_chars = lines[loc][:2]
```

Comment:
this code is very difficult to understand because you're trying to be too clever to "DRY".  there are two very simple but disparate cases -- I would write them separately so it is easy to understand instead of having to hack together whatever is happening here

Reply:
It felt like a lot of duplication when the code is so similar, but I've changed it.